### ISSUE TITLE
[SDK44] ExpoCamera/ExpoFaceDetector crashes app on Android

### ISSUE THREAD

**Issue by yonitou** [expo/expo#15830]
### Summary

Hello :) 

We have a managed app on SDK 44. 

We have a login screen using a expo-camera package to scan a BarCode. When being on this screen, if we send the app to the background and to the foreground again on Android, the app crashes and we have the following error : 

```
java.lang.NullPointerException: Parameter specified as non-null is null: method kotlin.jvm.internal.Intrinsics.checkNotNullParameter, parameter settings
	at expo.modules.facedetector.ExpoFaceDetector.setSettings(Unknown Source:2)
	at expo.modules.camera.ExpoCameraView.onHostResume(ExpoCameraView.kt:216)
	at expo.modules.adapters.react.services.UIManagerModuleWrapper$3.onHostResume(UIManagerModuleWrapper.java:127)
	at com.facebook.react.bridge.ReactContext.onHostResume(ReactContext.java:240)
	at com.facebook.react.ReactInstanceManager.moveToResumedLifecycleState(ReactInstanceManager.java:719)
	at com.facebook.react.ReactInstanceManager.onHostResume(ReactInstanceManager.java:623)
	at com.facebook.react.ReactInstanceManager.onHostResume(ReactInstanceManager.java:580)
	at com.facebook.react.ReactDelegate.onHostResume(ReactDelegate.java:53)
	at com.facebook.react.ReactActivityDelegate.onResume(ReactActivityDelegate.java:100)
```

We are not using `expo-barcode-scanner` or `expo-face-detector` package at all. 

FYI, we had no issues on SDK 43.

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

Android

### SDK Version (managed workflow only)

44

### Environment

```
 Expo CLI 5.0.1 environment info:
    System:
      OS: macOS 12.1
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 16.13.1 - ~/.nvm/versions/node/v16.13.1/bin/node
      npm: 8.3.0 - ~/.nvm/versions/node/v16.13.1/bin/npm
    Managers:
      CocoaPods: 1.11.2 - /Users/yonitouboul/.rbenv/shims/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 21.2, iOS 15.2, macOS 12.1, tvOS 15.2, watchOS 8.3
    IDEs:
      Android Studio: 2020.3 AI-203.7717.56.2031.7935034
      Xcode: 13.2.1/13C100 - /usr/bin/xcodebuild
    npmPackages:
      babel-preset-expo: 9.0.1 => 9.0.1 
      expo: ^44.0.0 => 44.0.1 
      react: 17.0.1 => 17.0.1 
      react-dom: 17.0.1 => 17.0.1 
      react-native: 0.64.3 => 0.64.3 
    npmGlobalPackages:
      eas-cli: 0.42.4
      expo-cli: 5.0.1
    Expo Workflow: managed
```

### Reproducible demo

- Go on a screen using `expo-camera` package on an Android smartphone
- Put the app in background 
- Wait for 3 seconds
- Put the app in foreground again
- Error triggered

**Comment by byCedric** [comment]
Hi @yonitou! Can you provide a small working repository where you run into this crash? That would help us debug what's going on.

**Comment by yonitou** [comment]
Hi @byCedric , thanks for your answer :) I will try to do so since our repo is quite big with sensitive data. 

Meanwhile, you can have a look at this issue that is pretty close to what I'm experiencing : https://github.com/expo/expo/issues/15547#issuecomment-1008799428

**Comment by paulocappa** [comment]
Up! I'm facing same issue here.

I'm not using `expo-face-detector` or `expo-barcode-scanner` and when I try to record a video it crashes with the above error. Also occurs when getting an image from `expo-image-picker`(launchImageLibraryAsync) 

I have a single page that only renders the camera.

**Comment by nikodunk** [comment]
Hey there! We're getting the same here. It also doesn't happen on the Expo client - only on the EAS built app. Doesn't happen on SDK 43 on either. Thanks!

**Comment by martinezleoml** [comment]
Hi there! We're getting the same errors on a bare React Native app on SDK 44. We're neither using `expo-face-detector` nor `expo-barcode-scanner` features, only the basic camera features.

**Comment by byCedric** [comment]
If anyone can supply the repro, we can debug this issue:
> Can you provide a small working repository where you run into this crash? That would help us debug what's going on.

**Comment by ivbrajkovic** [comment]
Repo can be gound here, it is the same isdue #15547

**Comment by NorthBlue333** [comment]
As #15547 is fixed, I tried to update to version `12.1.2` and it fixes the issue. Just letting you know :tada:

**Comment by giautm** [comment]
I'm running into the same crash on Android, `expo-camera@12.1.2`
<img width="870" alt="Screen Shot 2022-03-07 at 14 23 13" src="https://user-images.githubusercontent.com/12751435/156986181-d1461f36-831e-46ab-9b2d-c2556f1f7a3d.png">

https://github.com/expo/expo/blob/ce7ba16586229337c47c87a78f4401371f077662/android/versioned-abis/expoview-abi44_0_0/src/main/java/abi44_0_0/expo/modules/camera/ExpoCameraView.kt#L213-L218
